presto-native-execution/presto_cpp/main/http/HttpClient.cpp

```diff
@@ -50,13 +53,13 @@ void HttpResponse::append(std::unique_ptr<folly::IOBuf>&& iobuf) {
   VELOX_CHECK(!iobuf->isChained());
   uint64_t dataLength = iobuf->length();
 
-  void* buf = allocator_->allocateBytes(dataLength);
-  if (buf == nullptr) {
-    VELOX_FAIL(
-        "Cannot spare enough system memory to receive more HTTP response.");
+  if (pool_) {
+    void* buf = pool_->allocate(dataLength);
```

Comment:
MemoryPool::allocate() will not return nullptr, but instead throw if not enough memory, right?

Reply:
We throw instead of return nullptr. That's the intended behavior if there is not bug.